distributed/shuffle/tests/test_graph.py

```diff
@@ -46,6 +46,7 @@ def test_shuffle_helper(client: Client):
 
 def test_basic(client: Client):
     df = dd.demo.make_timeseries(freq="15D", partition_freq="30D")
+    df["name"] = df["name"].astype("string[python]")
```

Comment:
```suggestion

@pytest.mark.parametrize("cast_string", [True, False])
def test_basic(client: Client, cast_string: bool):    
    df = dd.demo.make_timeseries(freq="15D", partition_freq="30D")
    if cast_string:
        df["name"] = df["name"].astype("string[python]")

Reply:
Yeah, this is a live issue.  Arrow likes converting things to strings if they are actually strings.  I don't mind this.  I've been avoiding this for now, but it's good that you brought it up so that we can address it explicitly.  

I also haven't tested non-string object types at all.  I should add a larger test for this behavior.